org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/text/java/ContentAssistProcessor.java

```diff
@@ -730,4 +748,46 @@ public ICompletionProposal getSelectedProposal() {
 		return fSelectedProposal;
 	}
 
+	@Override
+	public boolean isCompletionProposalAutoActivation(char c, ITextViewer viewer, int offset) {
+		if(fAutoActivateOnType && Character.isLetterOrDigit(c)) {
+			return true;
+		}
+
+		char[] triggers= getCompletionProposalAutoActivationCharacters();
+		if (triggers != null) {
+			for (char trigger : triggers) {
+				if (c == trigger) {
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+	@Override
+	public boolean isContextInformationAutoActivation(char c, ITextViewer viewer, int offset) {
+		char[] triggers= getContextInformationAutoActivationCharacters();
+		if (triggers != null) {
+			for (char trigger : triggers) {
+				if (c == trigger) {
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+	public final void setAutoActivateOnType(boolean autoActivateOnType) {
+		fAutoActivateOnType= autoActivateOnType;
+	}
+
+	public final boolean isAutoActivateOnType() {
```

Comment:
I don't see it used anywhere. Is it expected?

Reply:
No this is just the getter that is added to comply with bean pattern, But i think we can remove these properties since it make sense to have it in the content assist it self rather than the processors to be beneficial in other places.